File: backend/mallapi/src/main/java/com/nareun/service/ProductService.java
package com.nareun.service;

import org.springframework.transaction.annotation.Transactional;

import com.nareun.dto.PageRequestDTO;
import com.nareun.dto.PageResponseDTO;
import com.nareun.dto.ProductDTO;

@Transactional
public interface ProductService {

  PageResponseDTO<ProductDTO> getList(PageRequestDTO pageRequestDTO); 

  Long register(ProductDTO productDTO);

  ProductDTO get(Long pno);

  void modify(ProductDTO productDTO);

  void remove(Long pno);

}

File: backend/mallapi-kotlin/src/main/kotlin/com/nareun/mallapikotlin/service/ProductService.kt
package com.nareun.mallapikotlin.service

import com.nareun.mallapikotlin.dto.PageRequestDTO
import com.nareun.mallapikotlin.dto.PageResponseDTO
import com.nareun.mallapikotlin.dto.ProductDTO
import org.springframework.transaction.annotation.Transactional

@Transactional
interface ProductService {
    fun getList(pageRequestDTO: PageRequestDTO) : PageResponseDTO<ProductDTO>

    fun register(productDTO: ProductDTO) : Long

    fun get(pno:Long) : ProductDTO

    fun modify(productDTO: ProductDTO)

    fun remove(pno: Long)
}